### DISCUSSION TITLE
G7 No BG - stuck in warmup - "Authorizing 2"

### DISCUSSION THREAD

**Discussion by rpm5099** [NightscoutFoundation/xDrip#3442] [Bug reports]
- Oneplus 6T
- Nightly Build from April 19 - xDrip-plus-20240419-645e8b4.apk

Switching from G6 to G7 for the first time.  Updated xDrip to the version above before starting.  Sensor was inserted.  At first I saw that it connected and started the sensor, but then I saw authorization failures and realized that I had (1) not removed the G6 transmitter from bluetooth history, (2) that transmitter may still have been in range, and (3) did not enter the transmitter id as I did not realize that's what it was.  After doing those things, I entered the transmitter ID I got a warmup message for 13 minutes, but then after that it stayed that way, xdrip displayed the warmup notice for several hours.  So I tried other things.

Did the following steps for the next 3 hours:
- Uninstalled and reinstalled xDrip multiple times, both reloading previous settings and starting clean
- Removed the device from bluetooth history and doing a "start sensor"
- rebooted the phone
- Tried to connect to the sensor on a different phone (Samsung S8) with the same results
- For all above followed the procedures listed [here](https://navid200.github.io/xDrip/docs/Dexcom/G7.html) *exactly* except for the build date of xDrip (used the latest nightly since the recommended was 2 weeks ago, and I know how is when things are changing)

It is my understanding that the G7 starts upon insertion, and that the only function of the "start sensor" button is to initiate a bluetooth connection to the sensor.  If that is correct, does the warm-up status come as a flag in the comms protocol from the sensor?  If authorization is successful but the data is very erratic will it be prevented from being transmitted from the sensor itself?

I would like to include all of the logs, but how do you export/save them from the app??  The only option I see is to email a developer - I appreciate the dedication but *usually* between the logs and looking at source code I can figure out what's going on (I would definitely post to help others out).  On that note, is there an "extra logging settings" tag for *all* modules?  If the option existed to save all until I get issues resolved I would.

Thanks to all developers of this project, it is truly appreciated - *this is the way*.

![Screenshot_20240428-232639](https://github.com/NightscoutFoundation/xDrip/assets/10326981/f0557d44-dacc-4ec9-bdd5-a70e90ffbe49)
![Screenshot_20240429-010336](https://github.com/NightscoutFoundation/xDrip/assets/10326981/550a0eb9-57ff-4ceb-9010-7af942e7694c)
![Screenshot_20240429-010344](https://github.com/NightscoutFoundation/xDrip/assets/10326981/f85663a0-c44c-4008-ad29-db75d9af27f8)
![Screenshot_20240429-010400](https://github.com/NightscoutFoundation/xDrip/assets/10326981/62889dbf-2dfe-4719-9142-32ca860d60bc)

**Comment by Navid200** [comment]
If readings are not valid from the transmitter perspective, it will show sensor errors.  Such errors may resolve eventually and readings may come back.  Or, they may not.
There are cases where the sensor is faulty and the filament never gets inserted.  
But, in the case of sensor errors, there will be connectivity and battery voltages will show.

What you are showing now in the status page looks like a connectivity issue.
In order to resolve a bad connectivity issue, I have had to uninstall xDrip, forget the Bluetooth device in the phone Android settings, install it fresh and not upload previous backups, set it up to pair to the device and it started showing readings.  Then, I have loaded the backups and that point.

Unfortunately, I have still not added the ability to save detail logs locally.
Currently, you can follow the instructions to upload detail logs for the lead developer to see:
https://navid200.github.io/xDrip/docs/Dexcom/OB1Debug.html

Do you happen to have a Dexcom G7 receiver?
Is your phone compatible with the Dexcom app?

**Comment by rpm5099** [comment]
Thanks for responding.  As you can see from the screenshots I *believe* I have the extra logging settings enabled, and I probably emailed them several times while attempting to export them to myself.  I never saw any battery voltages, but it clearly looked like it wsa connecting as there was bytes of data sent and received.  I no longer have those in my log after uninstalling and restoring the previous backup.

This phone (Oneplus 6t) does *not* have the Dexcom G7 app, however I tried it on a different phone and it also was not able to get any data from the sensor.  I do not have the Dexcom G7 receiver.

I am going to try again with a new sensor tonight and connecting to it with xDrip.  I am hoping that this sensor was just defective, but that would be pretty unlucky for the very first sensor.  I have not researched in great detail, but since the G7 does not require external activation (since it activates upon insertion) and based on the fact that people seem to freely switch which app/phone they are using to connect to it, I assumed that it does not become "bonded" to a single app or device or disable itself as a result of communication errors.  As long as two apps/devices are not interfering with each other what matters is that the transmitter ID is correct - is that accurate?

I'll write back with the results of what happens and whatever screenshots I can get.  If it fails I'll try to upload the logs and reference the url for this discussion.

Thanks again.

**Comment by rpm5099** [comment]
Thankfully the second sensor started up with no problems.  But the interesting thing is that the old sensor, after it was removed and in the trash, suddenly connected to the other phone with the Dexcom G7 app and sent the sensor data for the previous 12 hours.  I have to assume something was wrong with that sensor and manipulating it during removal somehow made it finally connect.  I had already started the new sensor on the actual phone so I did not check to see if it would connect to xdrip.  I uploaded the logs via xdrip covering the past 24 hours in case that is helpful at all.  I am curious know if something I did temporarily bricked the sensor.  

Once again thank you, I will go ahead and close the discussion.

Screenshots from dexcom g7 app from the sensor in the trash:

![from_trash1](https://github.com/NightscoutFoundation/xDrip/assets/10326981/b1abdeab-5196-4992-871d-7d1bd5bc5749)
![from_trash2](https://github.com/NightscoutFoundation/xDrip/assets/10326981/92f363fc-7015-442c-8fc1-f5009e9f4787)

**Reply by Navid200** [reply]
I'm glad it is working for you now.

**Comment by Navid200** [comment]
I don't know what just happened.  I had typed so much and was still typing and all of a sudden everything vanished and the discussion closed!

**Reply by rpm5099** [reply]
Shoot, I'm sorry, would like to know what you were saying...

**Reply by Navid200** [reply]
No worries.  Nothing that is not already in the guides.
I was answering your questions.  It's not a good idea to type a long message into someone else's thread.  That's a fact and not your fault.
